vitrivr-engine-module-pgvector/src/main/kotlin/org/vitrivr/engine/database/pgvector/descriptor/AbstractDescriptorReader.kt

```diff
@@ -0,0 +1,225 @@
+package org.vitrivr.engine.database.pgvector.descriptor
+
+import org.vitrivr.engine.core.database.descriptor.DescriptorReader
+import org.vitrivr.engine.core.model.descriptor.Descriptor
+import org.vitrivr.engine.core.model.descriptor.DescriptorId
+import org.vitrivr.engine.core.model.metamodel.Schema
+import org.vitrivr.engine.core.model.query.Query
+import org.vitrivr.engine.core.model.retrievable.RetrievableId
+import org.vitrivr.engine.core.model.retrievable.Retrieved
+import org.vitrivr.engine.database.pgvector.*
+import java.sql.ResultSet
+import java.util.*
+
+/**
+ * An abstract implementation of a [DescriptorReader] for Cottontail DB.
+ *
+ * @author Ralph Gasser
+ * @version 1.0.0
+ */
+abstract class AbstractDescriptorReader<D : Descriptor>(final override val field: Schema.Field<*, D>, override val connection: PgVectorConnection) : DescriptorReader<D> {
+
+    /** The name of the table backing this [AbstractDescriptorReader]. */
+    protected val tableName: String = "${DESCRIPTOR_ENTITY_PREFIX}_${this.field.fieldName}"
```

Comment:
`fieldName` should probably be sanitized to avoid possible SQL errors

Reply:
Not quite sure how to sanitize this.

Reply:
`[A-Za-z0-9]` or something?

Reply:
That would be more restrictive than what PostgreSQL can do.

However, if we restrict this, shouldn't the restriction take place one level above in the `Schema.Field` implementation?

Reply:
Yes, that would probably be better. I guess some sanitation could be applied generally, and if it is more restrictive than what the underlying layer supports but is still sufficient for the use case and easier to enforce, then I'd go with that. If we support arbitrary strings, this causes all kinds of problems while also not really providing any benefits. I don't see a reason, for example, to allow a field called "; DROP DATABASE vitrivr" or anything like that 😜

Reply:
Added this in the `Schema.Field` implementationt.